From 73767d1779b6c75853d48cb071d2854ad562ddce Mon Sep 17 00:00:00 2001
From: ztc1997 <dev@ztc1997.com>
Date: Mon, 13 May 2024 16:53:01 +0800
Subject: [PATCH] drm: Stub out debug prints

Within the display server process, __drm_dbg consumes significant CPU time:
    2.40%  [kernel]       [k] __drm_dbg

Instead of compiling in all DRM debug print statements, stub them out to
reduce runtime overhead and size.

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>
---
 drivers/gpu/drm/drm_print.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_print.c b/drivers/gpu/drm/drm_print.c
index 20f5e497b160..e6017cde2ecc 100644
--- a/drivers/gpu/drm/drm_print.c
+++ b/drivers/gpu/drm/drm_print.c
@@ -163,10 +163,14 @@ void __drm_printfn_info(struct drm_printer *p, struct va_format *vaf)
 }
 EXPORT_SYMBOL(__drm_printfn_info);
 
+#ifdef CONFIG_DRM_DEBUG_PRINT
 void __drm_printfn_debug(struct drm_printer *p, struct va_format *vaf)
 {
 	pr_debug("%s %pV", p->prefix, vaf);
 }
+#else
+void __drm_printfn_debug(struct drm_printer *p, struct va_format *vaf) {}
+#endif
 EXPORT_SYMBOL(__drm_printfn_debug);
 
 void __drm_printfn_err(struct drm_printer *p, struct va_format *vaf)
@@ -259,6 +263,7 @@ void drm_dev_printk(const struct device *dev, const char *level,
 }
 EXPORT_SYMBOL(drm_dev_printk);
 
+#ifdef CONFIG_DRM_DEBUG_PRINT
 void drm_dev_dbg(const struct device *dev, enum drm_debug_category category,
 		 const char *format, ...)
 {
@@ -281,7 +286,6 @@ void drm_dev_dbg(const struct device *dev, enum drm_debug_category category,
 
 	va_end(args);
 }
-EXPORT_SYMBOL(drm_dev_dbg);
 
 void __drm_dbg(enum drm_debug_category category, const char *format, ...)
 {
@@ -300,6 +304,12 @@ void __drm_dbg(enum drm_debug_category category, const char *format, ...)
 
 	va_end(args);
 }
+#else
+void drm_dev_dbg(const struct device *dev, enum drm_debug_category category,
+		 const char *format, ...) { }
+void __drm_dbg(enum drm_debug_category category, const char *format, ...) { }
+#endif
+EXPORT_SYMBOL(drm_dev_dbg);
 EXPORT_SYMBOL(__drm_dbg);
 
 void __drm_err(const char *format, ...)
